### DISCUSSION TITLE
Why does the iterative method not converge?

### DISCUSSION THREAD

**Discussion by nanguaxiaofendui** [idaholab/moose#25029] [Q&A General]
‘’’
l_clad = 1
r_in = 0.004
r_out = 0.005

[GlobalParams]
  displacements = 'disp_x disp_y disp_z'
  incremental = true
  family=LAGRANGE
  order=second
[]

[Mesh]
   [top_half_1]
    type = ConcentricCircleMeshGenerator
    num_sectors = 16
    radii = '${fparse r_in} ${fparse r_out}'
    rings = '1 3'
    has_outer_square = off
    pitch = 2
    portion =  right_half
    preserve_volumes = off
   []
   [top_half_2]
     type = BlockDeletionGenerator
     input = top_half_1
     block = '1'
     new_boundary = 'inner'
   []
   [clad_3d]
    type = AdvancedExtruderGenerator
    input = top_half_2
    heights = '${fparse l_clad }'
    direction = '0 0 1'
    num_layers = '8'
    bottom_boundary = '201'
    top_boundary = '202'
  []
  [node1]
    type = ExtraNodesetGenerator
    coord = '0.0 ${fparse -r_out} 0.0;'
    new_boundary = '21'
    input = clad_3d
  []
  [node2]
    type = ExtraNodesetGenerator
    coord = '0.0 ${fparse -r_out} ${fparse l_clad}'
    new_boundary = '22'
    input = node1
  []
  second_order=true
[]
[Variables]
  [./disp_x]
  [../]
  [./disp_y]
  [../]
  [./disp_z]
  [../]
[]

[Kernels]
  [gravity_y]
    type = Gravity
    variable = disp_y
    value = -9.81
  []
[]

[Modules/TensorMechanics/Master]
  [all]
    strain = FINITE
    add_variables = true
    generate_output ="vonmises_stress"
    cylindrical_axis_point1 = '0 0 0'
    cylindrical_axis_point2 = '0 0 1'
  []
[]

[BCs]
  [symmx]
    type = DirichletBC #
    variable = disp_x
    boundary = "21 22"
    value = 0
  []
  [symmy]
    type = DirichletBC  #
    variable = disp_y
    boundary = "21 22"
    value = 0
  []
  [symmz]
    type = DirichletBC  #
    variable = disp_z
    boundary = "21"
    value = 0
  []
  [sym]
    type = DirichletBC 
    variable = disp_x
    boundary = "left"
    value = 0
  []
  [Pressure]
    [outer_pressure] # 
      boundary = outer
      function = '4e6*t'
    []
    [inner_pressure] #
      boundary = inner
      function = '7e6*t'
    []
  []
[]

[Materials]
  [elasticity_tensor]
    type = ComputeIsotropicElasticityTensor
    youngs_modulus =7.5e10
    poissons_ratio = 0.3
  []
  [cstress]
    type = ComputeFiniteStrainElasticStress
  []
  [./cladding_density]
    type = Density
    density = 6551.0
  [../]
[]

[Executioner]
  type = Transient
  solve_type = 'NEWTON'
  line_search = 'none'
  automatic_scaling = true
  compute_scaling_once = false
[]
‘’’
The above example can be convergent with lu, but gmres cannot, why is that? Please tell me, thank you!

**Comment by GiudGiud** [comment]
Hello

not all solve methods will solve all nonlinear problems with any preconditioning. Please add a Preconditioning block and search for what works for your problem. The litterature in your field is usually a good starting point 

Guillaume

**Reply by nanguaxiaofendui** [reply]
Thank you very much

**Reply by GiudGiud** [reply]
Examples in moose can help as well btw, especially if we are solving the same physics
